File: swarmmanagercontroller/src/main/java/com/swarmmanager/docker/api/swarm/parameters/SwarmUpdateParameters.java
package com.swarmmanager.docker.api.swarm.parameters;

import com.swarmmanager.docker.api.common.parameters.QueryParameters;
import com.swarmmanager.docker.api.common.parameters.RequestBodyParameter;
import com.swarmmanager.docker.api.common.json.SwarmSpecJson;
import com.swarmmanager.rest.QueryParam;

import java.util.ArrayList;
import java.util.List;

public class SwarmUpdateParameters implements RequestBodyParameter, QueryParameters {

    private static final String VERSION_NAME = "version";

    private static final String ROTATE_WORKER_TOKEN_NAME = "rotateWorkerToken";

    private static final String ROTATE_MANAGER_TOKEN_NAME = "rotateManagerToken";

    private static final String ROTATE_MANAGER_UNLOCK_KEY_NAME = "rotateManagerUnlockKey";

    private QueryParam versionQueryParam;

    private SwarmSpecJson spec;

    private QueryParam rotateWorkerTokenQueryParam;

    private QueryParam rotateManagerTokenQueryParam;

    private QueryParam rotateManagerUnlockKeyQueryParam;

    public SwarmUpdateParameters() {
        spec = new SwarmSpecJson();
        versionQueryParam = new QueryParam(VERSION_NAME, 0L);
        rotateWorkerTokenQueryParam = new QueryParam(ROTATE_WORKER_TOKEN_NAME,false);
        rotateManagerTokenQueryParam = new QueryParam(ROTATE_MANAGER_TOKEN_NAME, false);
        rotateManagerUnlockKeyQueryParam = new QueryParam(ROTATE_MANAGER_UNLOCK_KEY_NAME, false);
    }

    public QueryParam getVersionQueryParam() {
        return versionQueryParam;
    }

    public SwarmUpdateParameters setVersionQueryParam(long versionValue) {
        this.versionQueryParam = new QueryParam(VERSION_NAME, versionValue);
        return this;
    }

    public SwarmUpdateParameters setSpec(SwarmSpecJson spec) {
        this.spec = spec;
        return this;
    }

    public SwarmUpdateParameters setRotateWorkerTokenQueryParam(boolean rotateWorkerTokenValue) {
        this.rotateWorkerTokenQueryParam = new QueryParam(ROTATE_WORKER_TOKEN_NAME, rotateWorkerTokenValue);
        return this;
    }

    public SwarmUpdateParameters setRotateManagerTokenQueryParam(boolean rotateManagerTokenValue) {
        this.rotateManagerTokenQueryParam = new QueryParam(ROTATE_MANAGER_TOKEN_NAME, rotateManagerTokenValue);
        return this;
    }

    public SwarmUpdateParameters setRotateManagerUnlockKeyQueryParam(boolean rotateManagerUnlockKeyValue) {
        this.rotateManagerUnlockKeyQueryParam = new QueryParam(ROTATE_MANAGER_UNLOCK_KEY_NAME, rotateManagerUnlockKeyValue);
        return this;
    }

    @Override
    public Object getRequestBody() {
        return spec;
    }

    @Override
    public List<QueryParam> getQueryParams() {
        List<QueryParam> queryParams = new ArrayList<>();
        queryParams.add(versionQueryParam);
        queryParams.add(rotateWorkerTokenQueryParam);
        queryParams.add(rotateManagerTokenQueryParam);
        queryParams.add(rotateManagerUnlockKeyQueryParam);
        return queryParams;
    }
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/swarm/parameters/SwarmUpdateParameters.kt
package com.swarmbit.docker.api.swarm.parameters

import com.swarmbit.docker.api.common.parameters.QueryParameters
import com.swarmbit.docker.api.common.parameters.RequestBodyParameter
import com.swarmbit.docker.api.common.rest.model.QueryParam
import com.swarmbit.docker.api.swarm.model.SwarmSpec

class SwarmUpdateParameters : RequestBodyParameter, QueryParameters {

    companion object {
        private const val VERSION_NAME = "version"
        private const val ROTATE_WORKER_TOKEN_NAME = "rotateWorkerToken"
        private const val ROTATE_MANAGER_TOKEN_NAME = "rotateManagerToken"
        private const val ROTATE_MANAGER_UNLOCK_KEY_NAME = "rotateManagerUnlockKey"
    }

    private var spec = SwarmSpec()
    private var versionQueryParam = QueryParam(VERSION_NAME, 0L)
    private var rotateWorkerTokenQueryParam = QueryParam(ROTATE_WORKER_TOKEN_NAME, false)
    private var rotateManagerTokenQueryParam = QueryParam(ROTATE_MANAGER_TOKEN_NAME, false)
    private var rotateManagerUnlockKeyQueryParam = QueryParam(ROTATE_MANAGER_UNLOCK_KEY_NAME, false)

    fun getVersionQueryParam(): QueryParam {
        return versionQueryParam
    }

    fun setVersionQueryParam(versionValue: Long): SwarmUpdateParameters {
        versionQueryParam = QueryParam(VERSION_NAME, versionValue)
        return this
    }

    fun setSpec(spec: SwarmSpec): SwarmUpdateParameters {
        this.spec = spec
        return this
    }

    fun setRotateWorkerTokenQueryParam(rotateWorkerTokenValue: Boolean): SwarmUpdateParameters {
        rotateWorkerTokenQueryParam = QueryParam(ROTATE_WORKER_TOKEN_NAME, rotateWorkerTokenValue)
        return this
    }

    fun setRotateManagerTokenQueryParam(rotateManagerTokenValue: Boolean): SwarmUpdateParameters {
        rotateManagerTokenQueryParam = QueryParam(ROTATE_MANAGER_TOKEN_NAME, rotateManagerTokenValue)
        return this
    }

    fun setRotateManagerUnlockKeyQueryParam(rotateManagerUnlockKeyValue: Boolean): SwarmUpdateParameters {
        rotateManagerUnlockKeyQueryParam = QueryParam(ROTATE_MANAGER_UNLOCK_KEY_NAME, rotateManagerUnlockKeyValue)
        return this
    }

    override val requestBody: Any
        get() = spec

    override val queryParams: List<QueryParam>
        get() {
            return listOf(
                versionQueryParam,
                rotateWorkerTokenQueryParam,
                rotateManagerTokenQueryParam,
                rotateManagerUnlockKeyQueryParam
            )
        }

}
